Adjusts salt.utils.stringutils.is_binary detection

piterpunk opened the pull request:
### What does this PR do?
- Deduplicates binary or text detection in `salt.utils.files.is_text` and `salt.utils.stringutils.is_binary`
- Updates `salt.utils.files.is_text` and `salt.utils.files.is_binary` to use the `salt.utils.stringutils.is_binary` to do the text or binary identification.

### What issues does this PR fix or reference?
Fixes #66706 
Fixes #62214

### Previous Behavior
The `salt.utils.files.is_text` and `salt.utils.files.is_binary` both gives false results when an utf-8 multibyte character is truncated. Different heuristics between the two modules also gives inconsistent results between the two methods.

### New Behavior
`salt.utils.files.is_text` and `salt.utils.files.is_binary` now uses the `salt.utils.stringutils.is_binary` to find if the given file is text or binary. Also, the detection was changed to accept truncated utf-8 multibyte characters.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Commit:
Adjusts utils.stringutils.is_binary detection

- Deduplicates binary or text detection in utils.files.is_text and
  utils.stringutils.is_binary
- Updates utils.files.is_text and utils.files.is_binary to use the
  utils.stringutils.is_binary to do the text or binary identification.

bdrx312 commented on salt/utils/stringutils.py:
```diff
@@ -181,31 +181,36 @@ def is_binary(data):
+    def int2byte(x):
+        return bytes((x,))
+
+    text_characters = b"".join(int2byte(i) for i in range(32, 127)) + b"\n\r\t\f\b"
```
Also I don't think there should be 2 separate assignments for this:

```python
text_characters = (
    b"".join(int2byte(i) for i in range(32, 127)) + b"\n\r\t\f\b" 
    + b"".join(
            int2byte(i) for i in range(128, 255)
        ))
```

piterpunk replied:
Ok, I will move the `text_characters` definition after the first checks.

I also agree that a comment will help to understand those ranges.

The first range defines the ASCII text characters. It excludes most control characters (values 0 to 31, but includes \n, \r, \t, \f and \b which are control characters common in texts) and all characters with values after 127. This is the original `text_characters` in `salt.utils.files.is_text` and, with the exception of `\f`, the one in `salt.utils.stringutils.is_binary`.

The UTF-8 encodes the non-ASCII characters as multibyte characters, which are represented by two or more bytes with values after 127 (128 to 255). So, after the byte array is decoded as a valid UTF-8 string, this range is also `text_characters`. But the UTF-8 encoding does nothing with the control characters (values 0 to 31) and we want to detect a text with more than 30% of non-text characters as binary.

That is why, instead of only returns `False` after the success in UTF-8 decoding the byte array, we increment the `text_characters` range and continue the validation if there is enough text characters to be considered as text.

Commit:
Small changes to salt.utils.stringutils.is_binary

- Define `text_characters` only when it will be used
- Adds commentaries about the `text_characters` definition
- Fix a corner case if a utf-8 multibyte character is truncated and
  more than 30% of the byte array have values between 1 and 31.
